src/HTTP.jl

```diff
@@ -631,4 +631,9 @@ function Base.parse(::Type{T}, str::AbstractString)::T where T <: Message
     return m
 end
 
+# only run if precompiling pkgimages
+if VERSION >= v"1.9.0-0" && Base.JLOptions().use_pkgimages != 0 && ccall(:jl_generating_output, Cint, ()) == 1
```

Comment:
Yeesh; it'd be nice if PrecompileTools provided an API for all these conditions (`isprecompiling`?)